docs/389ds/design/audit-json-logging-design.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Audit JSON Logging Design"
+---
+
+# Audit JSON Logging
+----------------
+
+Overview
+--------
+
+In order to make the audit logging more consumable by standard parsing tool it will written in a JSON format instead of using a partial LDIF format which not LDAP compliant in its current form.
+
+While the old format, after a bit if data massaging, could be replayed using a LDAP client (e.g. ldapmodify), this is most likely not used by most customers.  Instead, customers are looking for actual audit information that can by consumed by well known log parsers.
+
+JSON Design
+------------
+
+```
+[
+    {
+        date: 20243457458475847Z
```

Comment:
This date is invalid :)
And it would be nice to use the same ISO 8601 format as `Date` object's `toJSON()` method, since it will be consumed by JSON parsers anyway. I.e.
```
console.log(new Date().toJSON());
> 2024-04-23T06:24:52.306Z

console.log(Date.parse("2024-04-23T06:24:52.306Z"));
> 1713853492306
``` 
It's also more human readable than LDAP generalized time.

Reply:
Well I don't think Date.toJSON() exists in C, but I'm sure there is something else I can do here

Reply:
`strftime` with `%FT%TZ` format should be enough I think.

Reply:
> `strftime` with `%FT%TZ` format should be enough I think.

Or, we take it further and make it configurable where the setting value would be the parameter to strftime.  We've gotten requests before to allow for different time formats

Reply:
> strftime with %FT%TZ format should be enough I think
Yes but be careful to take the gmtime rather than the localtime
> Or, we take it further and make it configurable where the setting value would be the parameter to strftime. 
> We've gotten requests before to allow for different time formats
I know but there is a big difference:
  The ascii output is intended to be directly readable by humans.
   The json output must be post-processed by a tools
So do not bother with the format and let the tools format the time as its writer wants. 
The only important thing, it that the tools must be easy to convert (which is the case  with ISO 8601 format)
If you allow to configure the format, it makes things harder as the tool will need to determine the timestamp format to decode it.